### PR TITLE
Allow equality tests on arrays if their inner types support it

### LINKED ISSUE #2235: Comparison support for Array Type
### Issue to be solved

Right now there is no any option to check or compare two arrays. 

I have tried to compare 2 arrays with `==` and `!=` operators but it shows following error 

``` cannot apply binary operation == to types: `[Address]`, `[Address]`  ```

This feature really helps developer to compare two arrays and build smart-contract accordingly.

Thanks

### Suggested Solution

_No response_

### PR DESCRIPTION
**Pull request by dreamsmasher**
Closes #2235

## Description

Turns out we already had this functionality implemented, but it was disallowed in the checker.
______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: docs/language/operators.md [hunk 1]
```diff
@@ -372,6 +383,13 @@ Comparison operators work with boolean and integer values.
   x != y  // is `false`
   ```
 
+  ```cadence
+  // Equality tests of arrays are possible if their inner types are equatable.
```
Comment (SupunS) [inline docs/language/operators.md:387]:
```suggestion
  // Inequality tests of arrays are possible if their inner types are equatable.
```

File: runtime/sema/type.go [hunk 2]
```diff
@@ -2153,9 +2153,9 @@ func (t *VariableSizedType) IsImportable(results map[*Member]bool) bool {
 	return t.Type.IsImportable(results)
 }
 
-func (*VariableSizedType) IsEquatable() bool {
+func (v *VariableSizedType) IsEquatable() bool {
 	// TODO:
-	return false
+	return v.Type.IsEquatable()
```
Comment (SupunS) [inline runtime/sema/type.go:2158]: Maybe also remove the TODOs here and in other places (e.g: in `(*ConstantSizedType) IsEquatable()`)

**Comment by dreamsmasher** [comment]
hmm, seems like dictionaries are also unequatable. I'll open another issue, it's a good first contribution to the cadence repo. The implementation is basically the same as this

File: docs/language/operators.md [hunk 3]
```diff
@@ -288,9 +288,9 @@ Logical operators work with the boolean values `true` and `false`.
 
 ## Comparison Operators
 
-Comparison operators work with boolean and integer values.
+Comparison operators work with boolean and integer values. Array types can be tested for equality if their inner types support it.
 
-- Equality: `==`, for booleans and integers
+- Equality: `==`, for booleans, integers, and arrays containing equatable values.
```
[the remark this answers is not in the document]
Comment (dreamsmasher) [reply docs/language/operators.md:293]: took a quick look and it's just booleans, integers, optionals, and arrays

**Review by SupunS — APPROVED** [review]
Looks good! 👌 

Would love to have some more tests for different scenarios.

File: runtime/tests/interpreter/interpreter_test.go [hunk 4]
```diff
@@ -597,6 +597,68 @@ func TestInterpretParameters(t *testing.T) {
...
+	testBooleanFunction(t, "fixed array [Int; 3] should not be unequal to itself", false, `
+		let xs: [Int; 3] = [1, 2, 3]
+		return xs != xs
+	`)
+
+	testBooleanFunction(t, "fixed array [[Int; 2]; 1] should equal itself", true, `
+		let xs: [[Int; 2]; 1] = [[42, 1337]]
+		return xs == xs
+	`)
+}
```
[the remark this answers is not in the document]
Comment (dreamsmasher) [reply runtime/tests/interpreter/interpreter_test.go:662]: gotchu 😎

File: docs/language/operators.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (turbolent) [reply docs/language/operators.md]: There are more, see `Type.IsEquatable`. For example, paths, addresses, characters, etc.

[the remark this answers is not in the document]
Comment (dreamsmasher) [reply docs/language/operators.md]: ahh wait my bad, I was only grepping for cases where `IsEquatable` recurses on the inner type. lemme fix

**Review by turbolent — COMMENTED** [review]
Looks good, but tests are failing

**Review by dsainati1 — APPROVED** [review]

**Review by turbolent — APPROVED** [review]

**Pull request merged**